File: frontend/src/models/Credit.ts
// import { Contract } from "./Contract";
// import { ThirdParty } from "./ThirdParty";

export interface Credit {
    id :number;
    creditId: string; // Identifiant du crédit
    creditType: string; // Type de crédit
    nominalAmount: number; // Montant nominal
    cumulativeDisbursement: number; // Déblocage cumulatif
    setupDate: string; // Date de mise en place
    firstInstallmentDate: string; // Date de la première échéance
    nominalRate: number; // Taux nominal
    rateNature: string; // Nature du taux
    installmentCount: number; // Nombre d'échéances
    deferredType: string; // Type de différé
    restructured: boolean; // Restructuré (oui/non)
    restructuringCount: number; // Nombre de restructurations
    creditStatus: string; // Statut du crédit
    constantInstallmentAmount: number; // Montant de l'échéance constante
    unpaidAmount: number; // Montant impayé
    insuranceAmount: number; // Montant de l'assurance
    triggeredInstallmentNumber: number; // Numéro d'échéance déclenchée
    openingDate: string; // Date d'ouverture
    modificationDate: string; // Date de modification
    lastStatusDate: string; // Date du dernier statut
    cumulativeRedemptionAmount: number; // Montant cumulé des rachats
    lastRedemptionDate: string; // Date du dernier rachat
    agency: string; // Agence
    manager: string; // Gestionnaire a modifier 99%
    contract:{
        id:number;
        contractId:string
    }; // Identifiant du contrat
    thirdParty: {
         id:number;
         thirdPartyId:string;
    }; // Tiers
  }
  

File: backend/recouvrex-services/src/main/java/com/recouvrex/process/model/Credit.java
package com.recouvrex.process.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDate;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Entity
@Table(name = "credit")
public class Credit extends BaseEntity {

    private String creditId; // Identifiant du crédit

    private String creditType; // Type de crédit
    private BigDecimal nominalAmount; // Montant nominal
    private BigDecimal cumulativeDisbursement; // Déblocage cumulatif
    private LocalDate setupDate; // Date de mise en place
    private LocalDate firstInstallmentDate; // Date de la première échéance
    private double nominalRate; // Taux nominal
    private String rateNature; // Nature du taux
    private int installmentCount; // Nombre d'échéances
    private String deferredType; // Type de différé
    private boolean restructured; // Restructuré (oui/non)
    private int restructuringCount; // Nombre de restructurations
    private String creditStatus; // Statut du crédit
    private BigDecimal constantInstallmentAmount; // Montant de l'échéance constante
    private BigDecimal unpaidAmount; // Montant impayé
    private BigDecimal insuranceAmount; // Montant de l'assurance
    private int triggeredInstallmentNumber; // Numéro d'échéance déclenchée
    private LocalDate openingDate; // Date d'ouverture
    private LocalDate modificationDate; // Date de modification
    private LocalDate lastStatusDate; // Date du dernier statut
    private BigDecimal cumulativeRedemptionAmount; // Montant cumulé des rachats
    private LocalDate lastRedemptionDate; // Date du dernier rachat
    private String agency; // Agence
    private String manager; // Gestionnaire

    @ManyToOne
    @JoinColumn(name = "contract_id")
    private Contract contract;

    @ManyToOne
    @JoinColumn(name = "thirdparty_id")
    private ThirdParty thirdParty;

    private boolean reminderSent; // Relance déjà envoyée ?
}
